Action.cpp

```diff
@@ -217,62 +210,68 @@ namespace gameController{
     }
 
     auto Move::checkForFoul() const -> gameModel::Foul {
-
-        // @ToDo: Fouls
-
-        // Keilen
+        // Ramming
         if (env->getPlayer(this->target).has_value() &&
             !env->arePlayerInSameTeam(*(env->getPlayer(this->target).value()), *(this->actor))) {
-            return  gameModel::Foul::Keilen;
+            return  gameModel::Foul::Ramming;
```

Comment:
Das sieht irgendwie komisch aus, wieso nicht gleich mit shared_ptr (zumindest ein Overload dafür)

Reply:
sorry ich check grade nicht was genau du damit meinst

Reply:
Zeile drüber, das shared_ptr dereferenzieren

Reply:
also sollen da dann alle darunter liegenden methoden auf shared pointer umgebaut werden oder was meinst? weil ein overload bringt doch da auch nicht direkt was, dann wird halt die konstruktion in ne andere funktion ausgelagert oder versteh ich das grade einfach falsch?

Reply:
Ein Overload würde das hält übersichtlicher machen, inwieweit alles auf shared_ptr sinn macht kann ich nicht beurteilen, @Timmifixedit was meinst du dazu?

Reply:
@jonas-merkle wenn mein pr mit den shared_ptr angenommen wird, gibts eh überall nur noch shared_ptr. Dann sollten wir das umziehen auf ptr

Reply:
@Timmifixedit also alle Funktionen auf std::shared_ptr<Player> umbauen?

Reply:
@jonas-merkle machen wir dann im nächsten pr